config/config-network.yaml

```diff
@@ -173,12 +173,34 @@ data:
     # Knative doesn't know about that otherwise.
     default-external-scheme: "http"
 
+    # internal-encryption is deprecated and replaced by internal-dataplane-trust and internal-controlplane-trust
     # internal-encryption indicates whether internal traffic is encrypted or not.
-    # If this is "true", the following traffic are encrypted:
-    #  - ingress to activator
-    #  - ingress to queue-proxy
-    #  - activator to queue-proxy
     #
     # NOTE: This flag is in an alpha state and is mostly here to enable internal testing
     #       for now. Use with caution.
     internal-encryption: "false"
+
+    # internal-dataplane-trust indicates the level of trust established in the knative data-plane.
+    # internal-dataplane-trust = "disabled" (the default) - uses no encryption for internal traffic
+    # Using any other value ensures that the following traffic is encrypted using TLS:
+    #  - ingress to activator
+    #  - ingress to queue-proxy
+    #  - activator to queue-proxy
+    #
+    # internal-dataplane-trust = "minimal" ensures data messages are encrypted, Kingress authenticate that the receiver is a Kserv
+    # internal-dataplane-trust = "enabled" same as "minimal" and in addition, Kingress authenticate that Kserv is at the correct namespace
+    # internal-dataplane-trust = "mutual" same as "enabled" and in addition, Kserv authenticate that the messages come from the Kingress
+    # internal-dataplane-trust = "identity" same as "mutual" with Kingress adding a trusted sender identity to the message
+    #
+    # NOTE: This flag is in an alpha state and is mostly here to enable internal testing for now. Use with caution.
+    internal-dataplane-trust: "disabled"
+
+    # internal-controlplane-trust indicates the level of trust established in the knative control-plane.
+    # internal-controlplane-trust = "disabled" (the default) - uses no encryption for internal traffic
+    # Using any other value ensures that control traffic is encrypted using TLS.
+    #
+    # internal-controlplane-trust = "enabled" ensures control messages are encrypted using TLS (client authenticate the server)
+    # internal-controlplane-trust = "mutual" ensures control messages are encrypted using mTLS (client and server authenticate each other)
+    #
+    # NOTE: This flag is in an alpha state and is mostly here to enable internal testing for now. Use with caution.
+    internal-controlplane-trust: "disabled"
```

Comment:
I think I understood about `internal-dataplane-trust` but I am still not sure how `internal-controlplane-trust` will be used. Could you share the link docs or github issue for your design/idea where I can refer to?

Reply:
@nak3 
AFAIK, we have little on this right now. 
We understand that we need to start building control plane trust (after we complete the more critical data plane trust) 
And we define a flag that will allow the introduction of such work moving forward.

The little we do have is now summarized in https://docs.google.com/document/d/19R0YOBAZfed2bl99-fdUnQ89cz3hoPY6KJNxhn6b40A/edit#heading=h.n8a530nnrb  - which is incomplete and need more contributions.